newjersey/navigator.business.nj.gov
# chore:[#186705729] add query find user by business name

## Description

This PR introduces a new feature to search for users by their business name using a query in AWS DynamoDB. The approach involves creating an SQL query statement in the DynamoUserDataClient that searches for users with businesses. The query checks for the existence of businesses and ensures that the businessName field is not missing.

Once the query is executed, the results are ingested into the DynamoDataClient. There, the business names are normalized (trimmed and converted to lowercase) to ensure case-insensitive matching when filtering for users with a business name that matches the search term.



### Ticket



This pull request resolves [#186705729](https://www.pivotaltracker.com/story/show/186705729).

### Approach

- Added `queryUsersWithBusinesses` in `DynamoUserDataClient` to execute a DynamoDB query that retrieves users with businesses.
- Implemented `findUserByBusinessName` in `DynamoDataClient`, which filters users based on a normalized, case-insensitive business name. 
- Also updated the Router tests where the mock/stub `DynamoDataClient` lives as well as the Batch files where the mock/stub DynamoUserDataClient lives. 



### Steps to Test
1.  Run Unit Tests
 The following tests were created to verify the functionality of the changes here and previous existing logic in the DynamoDataClient.
 
 - `should find user by businessName ...`: This test ensures that the findUserByBusinessName method filters users correctly by matching business names, including case-insensitive matches.
 - `should migrate data for multiple users with outdated versions...`: This test verifies the migration of outdated user data, ensuring the business data is updated and logged appropriately
 
 2. Run the test suite with `yarn test` and ensure the tests pass successfully. 
 


### Notes



## Code author checklist
